staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator.go

```diff
@@ -98,5 +110,17 @@ func (a *cachedTokenAuthenticator) AuthenticateToken(ctx context.Context, token
 }
 
 func keyFunc(auds []string, token string) string {
-	return fmt.Sprintf("%#v|%v", auds, token)
+	// individually hash each input to prevent ambiguities, i.e. "xy" + "z" == "x" + "yz"
+	audsHash := sha256.Sum256([]byte(fmt.Sprintf("%#v", auds)))
+	tokenHash := sha256.Sum256([]byte(fmt.Sprintf("%#v", token)))
+
+	// then hash all inputs with a random prefix
+	hash := sha256.New()
+
+	// Write() on hash never fails
+	_, _ = hash.Write(randomCachePrefix)
```

Comment:
No need for the assignment if you are going to ignore the return. However, I might check the errors just because.